seatunnel-core/seatunnel-starter/src/test/resources/args/user_defined_params.conf

```diff
@@ -0,0 +1,47 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+######
+###### This config file is a demonstration of streaming processing in seatunnel config
+######
+
+env {
+  # You can set engine configuration here
+  execution.parallelism = 1
+  job.mode = "BATCH"
+  checkpoint.interval = 5000
+  #execution.checkpoint.data-uri = "hdfs://localhost:9000/checkpoint"
+}
+
+source {
+  # This is a example source plugin **only for test and demonstrate the feature source plugin**
+  FakeSource {
+    result_table_name = ${fake_table}
```

Comment:
Test multiple variables instead of just one

Reply:
> Test multiple variables instead of just one

The test case has been modified to demonstrate a scenario with multiple variables

Reply:
code changed!
`env {
  # You can set engine configuration here
  execution.parallelism = 1
  job.mode = "BATCH"
  checkpoint.interval = 5000
  #execution.checkpoint.data-uri = "hdfs://localhost:9000/checkpoint"
}

source {
  # This is a example source plugin **only for test and demonstrate the feature source plugin**
  FakeSource {
    result_table_name = ${fake_source_table}
    parallelism = ${fake_parallelism}
    username = ${username}
    password = ${password}
    schema = {
      fields {
        name = "string"
        age = "int"
      }
    }
  }
}

sink {
  console {
     result_table_name = ${fake_sink_table}
     username = ${username}
     password = ${password}
  }
}`

`    @Test
    public void testUserDefinedParamsCommand() throws URISyntaxException {
        int fakeParallelism = 16;
        String username = "seatunnel=2.3.1";
        String password = "dsjr42=4wfskahdsd=w1chh";
        String fakeSourceTable = "fake";
        String fakeSinkTable = "sink";
        String[] args = {
            "-c",
            "/args/user_defined_params.conf",
            "-e",
            "local",
            "-i",
            "fake_source_table=" + fakeSourceTable,
            "-i",
            "fake_parallelism=" + fakeParallelism,
            "-i",
            "fake_sink_table=" + fakeSinkTable,
            "-i",
            "password=" + password,
            "-i",
            "username=" + username
        };
        ClientCommandArgs clientCommandArgs =
                CommandLineUtils.parse(args, new ClientCommandArgs(), "seatunnel-zeta", true);
        clientCommandArgs.buildCommand();
        URL resource = ClientCommandArgsTest.class.getResource("/args/user_defined_params.conf");

        Config config =
                ConfigFactory.parseFile(Paths.get(resource.toURI()).toFile())
                        .resolve(ConfigResolveOptions.defaults().setAllowUnresolved(true))
                        .resolveWith(
                                ConfigFactory.systemProperties(),
                                ConfigResolveOptions.defaults().setAllowUnresolved(true));
        List<? extends ConfigObject> sourceConfigs = config.getObjectList("source");
        for (ConfigObject configObject : sourceConfigs) {
            Config sourceConfig = configObject.toConfig();

            String tableName = sourceConfig.getString("result_table_name");
            Assertions.assertEquals(tableName, fakeSourceTable);

            int parallelism = Integer.parseInt(sourceConfig.getString("parallelism"));
            Assertions.assertEquals(fakeParallelism, parallelism);

            Assertions.assertEquals(sourceConfig.getString("username"), username);
            Assertions.assertEquals(sourceConfig.getString("password"), password);
        }
        List<? extends ConfigObject> sinkConfigs = config.getObjectList("sink");
        for (ConfigObject sinkObject : sinkConfigs) {
            Config sinkConfig = sinkObject.toConfig();
            String tableName = sinkConfig.getString("result_table_name");
            Assertions.assertEquals(tableName, fakeSinkTable);

            Assertions.assertEquals(sinkConfig.getString("username"), username);
            Assertions.assertEquals(sinkConfig.getString("password"), password);
        }
    }`

Reply:
```json
env {
  execution.parallelism = 1
  job.mode = "BATCH"
  checkpoint.interval = 5000
}

source {
  FakeSource {
    result_table_name = ${fake_source_table}
    parallelism = ${fake_parallelism}
    username = ${username}
    password = ${password}
    schema = {
      fields {
        name = "string"
        age = "int"
      }
    }
  }
}

sink {
  console {
     result_table_name = ${fake_sink_table}
     username = ${username}
     password = ${password}
  }
}
```
public void testUserDefinedParamsCommand() throws URISyntaxException {
        int fakeParallelism = 16;
        String username = "seatunnel=2.3.1";
        String password = "dsjr42=4wfskahdsd=w1chh";
        String fakeSourceTable = "fake";
        String fakeSinkTable = "sink";
        String[] args = {
            "-c",
            "/args/user_defined_params.conf",
            "-e",
            "local",
            "-i",
            "fake_source_table=" + fakeSourceTable,
            "-i",
            "fake_parallelism=" + fakeParallelism,
            "-i",
            "fake_sink_table=" + fakeSinkTable,
            "-i",
            "password=" + password,
            "-i",
            "username=" + username
        };
        ClientCommandArgs clientCommandArgs =
                CommandLineUtils.parse(args, new ClientCommandArgs(), "seatunnel-zeta", true);
        clientCommandArgs.buildCommand();
        URL resource = ClientCommandArgsTest.class.getResource("/args/user_defined_params.conf");

        Config config =
                ConfigFactory.parseFile(Paths.get(resource.toURI()).toFile())
                        .resolve(ConfigResolveOptions.defaults().setAllowUnresolved(true))
                        .resolveWith(
                                ConfigFactory.systemProperties(),
                                ConfigResolveOptions.defaults().setAllowUnresolved(true));
        List<? extends ConfigObject> sourceConfigs = config.getObjectList("source");
        for (ConfigObject configObject : sourceConfigs) {
            Config sourceConfig = configObject.toConfig();

            String tableName = sourceConfig.getString("result_table_name");
            Assertions.assertEquals(tableName, fakeSourceTable);

            int parallelism = Integer.parseInt(sourceConfig.getString("parallelism"));
            Assertions.assertEquals(fakeParallelism, parallelism);

            Assertions.assertEquals(sourceConfig.getString("username"), username);
            Assertions.assertEquals(sourceConfig.getString("password"), password);
        }
        List<? extends ConfigObject> sinkConfigs = config.getObjectList("sink");
        for (ConfigObject sinkObject : sinkConfigs) {
            Config sinkConfig = sinkObject.toConfig();
            String tableName = sinkConfig.getString("result_table_name");
            Assertions.assertEquals(tableName, fakeSinkTable);

            Assertions.assertEquals(sinkConfig.getString("username"), username);
            Assertions.assertEquals(sinkConfig.getString("password"), password);
        }
    }
```java

```

Reply:
`public void testUserDefinedParamsCommand() throws URISyntaxException {
        int fakeParallelism = 16;
        String username = "seatunnel=2.3.1";
        String password = "dsjr42=4wfskahdsd=w1chh";
        String fakeSourceTable = "fake";
        String fakeSinkTable = "sink";
        String[] args = {
            "-c",
            "/args/user_defined_params.conf",
            "-e",
            "local",
            "-i",
            "fake_source_table=" + fakeSourceTable,
            "-i",
            "fake_parallelism=" + fakeParallelism,
            "-i",
            "fake_sink_table=" + fakeSinkTable,
            "-i",
            "password=" + password,
            "-i",
            "username=" + username
        };
        ClientCommandArgs clientCommandArgs =
                CommandLineUtils.parse(args, new ClientCommandArgs(), "seatunnel-zeta", true);
        clientCommandArgs.buildCommand();
        URL resource = ClientCommandArgsTest.class.getResource("/args/user_defined_params.conf");

        Config config =
                ConfigFactory.parseFile(Paths.get(resource.toURI()).toFile())
                        .resolve(ConfigResolveOptions.defaults().setAllowUnresolved(true))
                        .resolveWith(
                                ConfigFactory.systemProperties(),
                                ConfigResolveOptions.defaults().setAllowUnresolved(true));
        List<? extends ConfigObject> sourceConfigs = config.getObjectList("source");
        for (ConfigObject configObject : sourceConfigs) {
            Config sourceConfig = configObject.toConfig();

            String tableName = sourceConfig.getString("result_table_name");
            Assertions.assertEquals(tableName, fakeSourceTable);

            int parallelism = Integer.parseInt(sourceConfig.getString("parallelism"));
            Assertions.assertEquals(fakeParallelism, parallelism);

            Assertions.assertEquals(sourceConfig.getString("username"), username);
            Assertions.assertEquals(sourceConfig.getString("password"), password);
        }
        List<? extends ConfigObject> sinkConfigs = config.getObjectList("sink");
        for (ConfigObject sinkObject : sinkConfigs) {
            Config sinkConfig = sinkObject.toConfig();
            String tableName = sinkConfig.getString("result_table_name");
            Assertions.assertEquals(tableName, fakeSinkTable);

            Assertions.assertEquals(sinkConfig.getString("username"), username);
            Assertions.assertEquals(sinkConfig.getString("password"), password);
        }
    }`